src/Codeception/Lib/Connector/Yii2.php

```diff
@@ -422,33 +423,20 @@ protected function encodeCookies(
      */
     protected function mockMailer(array $config): array
     {
-        // options that make sense for mailer mock
-        $allowedOptions = [
-            'htmlLayout',
-            'textLayout',
-            'messageConfig',
-            'messageClass',
-            'useFileTransport',
-            'fileTransportPath',
-            'fileTransportCallback',
-            'view',
-            'viewPath',
-        ];
-
         $mailerConfig = [
             'class' => TestMailer::class,
             'callback' => function (MessageInterface $message) {
                 $this->emails[] = $message;
-            }
+            },
+            'mailer' => [
+                'class' => Mailer::class,
```

Comment:
```suggestion
                'class' => Mailer::class,
                'transportFactory' => new NullTransportFactory()
```
I think this would solve my issues with the PR, but I did not test this; so will need a proper confirmation on this.